### ISSUE TITLE
bug: flakes import path breaks if specifying a path multiple files deep

### ISSUE THREAD

**Issue by selene57** [nix-community/home-manager#2495]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue description

When using flakes, either specifying a home configuration through a nixos module or through the homeConfigurations way, it does not allow for specifying a configuration import path that is within multiple subdirectories.

E.g. When I try to use "./users/selene/home.nix" it throws the following error:

[selene@nixos:~/nixfiles]$ home-manager switch
error: getting status of '/nix/store/jkl7jkdq7cr9hhrih927nbnjv2cxjgc5-source/users/selene/config': No such file or directory
(use '--show-trace' to show detailed location information)

When I try use "./home.nix" and place the file in the appropriate location (same directory as my flake.nix), then does not throw this error.

### System information

```markdown
[selene@nixos:~/nixfiles]$ nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.06 MiB download, 0.30 MiB unpacked):
  /nix/store/94n19nsi7xdsna40zl7iy4476kwldlzg-bash-interactive-5.1-p8-dev
copying path '/nix/store/94n19nsi7xdsna40zl7iy4476kwldlzg-bash-interactive-5.1-p8-dev' from 'https://cache.nixos.org'...
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.3, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4`
 - channels(selene): `""`
 - channels(root): `"home-manager, nixos-21.11pre329879.c5ed8beb478"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by terlar** [comment]
You said you import "./users/selene/**home.nix**"
But it complains about: '/nix/store/jkl7jkdq7cr9hhrih927nbnjv2cxjgc5-source/users/selene/**config**': No such file or directory

Do you have the mentioned config folder/file in your source?

**Comment by selene57** [comment]
I originally had a folder named config, but I got rid of it and the references to it in home.nix. trying to make things as simple as possible to just get flakes + home manager up and running. It was still giving me that error.

I ended up simplifying things and doing a clean install of my system. This stopped the errors I was receiving, so i don't know if it was just an artifact of how I had it set up before or an actual bug of some sort. I'm going to close this issue for now.

Thanks!
